openweather/src/main/java/pages/SearchPage.java

```diff
@@ -25,16 +28,40 @@ public SearchPage(WebDriver driver) {
     }
 
     // TODO: use a common method to get row/cell text
-    public List<CityInfo> getSearchResult() {
+    public List<CityInfo>  getSearchResult() {
         List<CityInfo> cities = new ArrayList<>();
-        for (WebElement row : getTableRows()) {
-            String name = row.findElement(By.tagName("a")).getText();
-            String temperature =
-                    row.findElement(By.xpath("//*[@class='badge badge-info'")).getText();
-            cities.add(new CityInfo(name, temperature));
-        }
+            for (WebElement row : getTableRows()) {
+                String name = row.findElement(By.tagName("a")).getText();
+                String temperature =
+                        row.findElement(By.xpath("//span[@class='badge badge-info']")).getText();
+                cities.add(new CityInfo(name, temperature));
+            }
 
         return cities;
+   }
+
+    public void checkSearchResult(String searchCity) {
```

Comment:
Page object (PO) is used for dealing with Page's element only, don't add validation method into this. Here is what I recommend:
I will write a validation class (validator) for SearchResult, SearchResultValidation.class for example. The class will be used for dealing with your validation logic.  In case you need "data" for validation, you will need to add more methods to PO. Shortly, PO is like a tool, it provides what you need on the fly, but not a final product.